src/GuardClauses/GuardAgainstNullExtensions.cs

```diff
@@ -79,6 +79,29 @@ public static partial class GuardClauseExtensions
             return input;
         }
 
+#if NET5_0_OR_GREATER
+        /// <summary>
+        /// Throws an <see cref="ArgumentException" /> if <paramref name="input" /> is an empty string.
+        /// </summary>
+        /// <param name="guardClause"></param>
+        /// <param name="input"></param>
+        /// <param name="parameterName"></param>
+        /// <param name="message">Optional. Custom error message</param>
+        /// <returns><paramref name="input" /> if the value is not an empty string.</returns>
+        /// <exception cref="ArgumentException"></exception>
+        public static ReadOnlySpan<char> NullOrEmpty(this IGuardClause guardClause,
+            ReadOnlySpan<char> input,
```

Comment:
I'd say make this nullable and have an overload for just `Empty`

Reply:
I guess when it comes to `Span<T>` you can even do something like:
```csharp
public static ReadOnlySpan<T>? NullOrEmpty<T>(this IGuardClause guardClause,
            ReadOnlySpan<T>? input,
            string parameterName,
            string? message = null)
      // ...
```
```csharp
public static ReadOnlySpan<T> Empty<T>(this IGuardClause guardClause,
            ReadOnlySpan<T> input,
            string parameterName,
            string? message = null)
      // ...
```

That way you can also just check if a `Span<int>` is empty for example

Reply:
Are you sure about nullability for ref structs? I tried to use `ReadOnlySpan<T>?` but get error:
`CS0306 The type 'ReadOnlySpan<char>' may not be used as a type argument`

Reply:
I just quickly tested these lines:
```csharp
 private void Test (ReadOnlySpan<char>? asdf) { }
 private ReadOnlySpan<T>? Test<T>(ReadOnlySpan<T>? asdf) { }
``` 
and it compiled.
Maybe nullability isn't on for this repo?

Reply:
Nullability is [enabled](https://github.com/ardalis/GuardClauses/blob/main/Directory.Build.props#L5)
Also, I reproduced it in an online compiler - https://dotnetfiddle.net/dxwtRA

Reply:
Whoops, so I tested the code in an open file which ended up being a generated file 🤦🏼 
I did some checks with the `[AllowNull]` and `[NotNull]` attributes but actually it makes no sense to support the nullability at all since you can't declare them as null anyways.